Add camera preview and checks before joining event

harish-aka-shivi opened the pull request:
Add the preview and checks
- [x] Ask permission
- [x] Add modal with camera stream
- [x] Improve UI
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

harish-aka-shivi committed:
feat: add camera preview in preview dialog

harish-aka-shivi committed:
feat: add redux logic

harish-aka-shivi committed:
feat: use firebase instead of store to track audio and vidoe mute

harish-aka-shivi committed:
feat: add wanring in case camera permission is not given

harish-aka-shivi committed:
feat: make join event text dynamic

harish-aka-shivi committed:
feat: add hooks to track video and audio in local storage

harish-aka-shivi committed:
feat: add option to join without permission

harish-aka-shivi committed:
fix: change color of lable to white

harish-aka-shivi committed:
fix: make availale lable white with opacity

harish-aka-shivi committed:
feat: add mic/video contols on topbar

harish-aka-shivi committed:
feat: add video setting in dropdown

This commit makes the tech check variabls via context
Add a useMedia stream hook. we are not using it for now

harish-aka-shivi committed:
fix: open video check when joining call

Technical check will be opened in networking container and confernence container if its a jitsi call.
Change the text when there is permission denied or dissmissed

harish-aka-shivi committed:
fix: add close button, title, subtitle in the modal

harish-aka-shivi committed:
feat: add device setting menu

harish-aka-shivi committed:
refactor: reabse and abstract provider component

harish-aka-shivi committed:
feat: refactor the context

- add audio input, video input and audio ouput states in context

harish-aka-shivi committed:
fix: fix bug due which toggles the audio and video related to each other

- Strange bug which was causing issues when toggling audio and video mute and including overriding the preferences.
The fix inclues having the separate values in localstorage for audio and videos

- Also add the use Jitsi hook

harish-aka-shivi committed:
fix: sycn device change state of jitsi player with context

harish-aka-shivi committed:
refactor: abstract jisti's logic to useJitsiHook

joaoaguiam commented on src/Hooks/useJitsi.js:
```diff
@@ -1,31 +1,43 @@
-  jitsiApi,
-  setJitsiApi,
-  callEndedCb,
+  domain="meet.jit.si",
```
This should not be hardcoded here, but instead, use a constant like the one we have on Modules/jitsi.js: DEFAULT_JITSI_SERVER

harish-aka-shivi replied:
Right. will do that

joaoaguiam commented on src/Hooks/useJitsi.js:
```diff
@@ -1,31 +1,43 @@
   displayName,
   sessionId,
-  conferenceVideoType,
+  conferenceVideoType="JITSI",
```
I am not sure that we should have the conferenceVideoType on this hook. 
We have two options:
- Remove the jitsi part from the conference container and create a new component just for the jitsi main stage that will call use jitsi
- Have a "enabled" variable on the useJitsi hook instead of this one

I would prefer the first option, it's cleaner. What do you think?

harish-aka-shivi replied:
Yes, I added a separate component for the main stage. I was trying it with networking area also but It was leading to reinitializing the component, so sticking with calling hooks in networking containers.

joaoaguiam commented on src/Hooks/useJitsi.js:
```diff
@@ -1,31 +1,43 @@
+      return;
+    }
 
     if (conferenceVideoType === "JITSI" && loaded && lastRoomLoaded !== roomName) {
```
I have changed this condition on the conference container because of the backstage feature. Can you please check the differences here: https://github.com/veertly/app/commit/42e1a09f9823c701aec6b79df1da6b613f1d15d3#diff-46c206b34a00a81456a8dc5c2612989d

harish-aka-shivi replied:
I don't understand this code
```
 if (eventSessionDetails.conferenceVideoType !== "JITSI" && loaded) {
      // dispose existing jitsi
      if (jitsiApi) {
        jitsiApi.executeCommand("hangup");
        jitsiApi.dispose();
        setLastRoomLoaded(null);
      }
    }
```

If we are not initializing jitsi when it is not of type Jitsi, how will it be true any time?

joaoaguiam replied:
This happened when one changes the type of event from jitsi to livestream on the cockpit, if the page is loaded, it then still keeps the jitsi component in the screen. You can try by running the cockpit and change back and forth the type of event.

harish-aka-shivi replied:
So, I handled differently. so when event type changes, I am disposing of the jitsiAPI instance. I tried setting values In the Cockpit and couldn't  reproduce it

joaoaguiam commented on src/Containers/EventSession/ConferenceRoomContainer.js:
```diff
@@ -106,87 +125,189 @@ export default () => {
+    callEndedCb: handleCallEnded,
+  });
+
+  // useEffect(() => {
```
Let's not keep this code commented out :P

joaoaguiam reviewed (changes requested):
@harish-aka-shivi Can you please merge this branch with the stage code? I feel there will be some conflicts. Please check the conference room container as I have changed some logic there.

harish-aka-shivi committed:
feat: add test sound

harish-aka-shivi commented on src/Hooks/useMuteAudioVideo.js:
```diff
@@ -0,0 +1,43 @@
+  // const [muteAudioInternal, setMuteAudioInternal] = useLocalStorage("vrtly_mute_audio", {});
+  // const [muteVideoInternal, setMuteVideoInternal] = useLocalStorage("vrtly_mute_video", {});
+  
+  const [muteAudioVideoState, setMuteAudioVideoState] = useLocalStorage("vrtly_mute_audio_video_state", {});
```
Yes, just tried keeping it short, Anyways need the key to be in the same format as others

harish-aka-shivi commented on src/Hooks/useShowTechCheck.js:
```diff
@@ -0,0 +1,94 @@
+}
+
+const useShowTechCheck = (sessionId) => {
+  const [techCheckState, setTechCheckState] = useLocalStorage("vrtly_show_tech_check", {});
```
Will fix it.

harish-aka-shivi committed:
refactor: update code according to Joao's suggestion

harish-aka-shivi commented on src/Containers/EventSession/ConferenceRoomContainer.js:
```diff
@@ -106,87 +125,189 @@ export default () => {
+  const prefixStr = prefix !== undefined ? `-${prefix}` : "";
+  const roomName = "veertly" + prefixStr + "-" + sessionId;
+  const domain = getJistiDomain(eventSessionDetails);
+  const subject = "Main Stage";
```
Done, 👍

hightechgrace replied:
Awesome

harish-aka-shivi committed:
fix: change main stage name according to cockpit name

- dispose the previous jitsi instance
- Fix bug when event type is changed from jitsi to facebook, so need to dispose appropriately

harish-aka-shivi committed:
fix: useMedia devices bug not giving devices when permissions are allowed

Render jitsi player in Networking room container using JitsiPlayerComponent
